examples/Entity_extraction_for_long_documents.ipynb

```diff
@@ -0,0 +1,370 @@
+    "print(template_prompt)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['1. How is a minor overspend breach calculated: Relevant Costs reported exceed the Power Unit Cost Cap by less than 5% (Page 24)',\n",
+       " '2. How is a major overspend breach calculated: Relevant Costs reported exceed the Power Unit Cost Cap by more than 5% (Page 24)',\n",
+       " '3. Which years do these financial regulations apply to: 2023-2026 (Page 2)',\n",
+       " '3. Which years do these financial regulations apply to: 2022 (Page 4)',\n",
+       " '3. Which years do these financial regulations apply to: 2022 (Page 8)',\n",
+       " '3. Which years do these financial regulations apply to: 2022 (Page 1)',\n",
+       " '3. Which years do these financial regulations apply to: 2022 (Page 1)',\n",
+       " '3.Which years do these financial regulations apply to: 2022 (Page 1)',\n",
+       " '3. Which years do these financial regulations apply to: 2023, 2024, 2025, 2026 onwards (Page 34)',\n",
+       " '3.Which years do these financial regulations apply to: 2022 (Page 1)',\n",
+       " '3.Which years do these financial regulations apply to: 2026 onwards (Page 41)',\n",
+       " '3.Which years do these financial regulations apply to: 2026 to 2030 seasons (inclusive) (Page 46)',\n",
+       " '3. Which years do these financial regulations apply to: 2022 (Page 1)',\n",
+       " '3. Which years do these financial regulations apply to: 2022 (Page 57)',\n",
+       " '3. Which years do these financial regulations apply to: 2022 (Page 61)']"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "chunks = []\n",
+    "results = []\n",
+    "\n",
+    "for i in range(0, len(clean_text), 10000):\n",
+    "    chunks.append(clean_text[i:i+10000])\n",
+    "\n",
+    "for chunk in chunks:\n",
+    "    results.append(extract_chunk(chunk,template_prompt))\n",
+    "    \n",
+    "groups = [r.split('\\n') for r in results]\n",
+    "\n",
+    "# zip the groups together\n",
+    "zipped = list(zip(*groups))\n",
+    "zipped = [x for y in zipped for x in y if \"Not specified\" not in x and \"__\" not in x]\n",
+    "zipped"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Consolidation\n",
+    "\n",
+    "We've been able to extract the first two answers safely, while the third was confounded by the date that appeared on every page, though the correct answer is in there as well.\n",
+    "\n",
+    "To tune this further you can consider experimenting with:\n",
+    "- A more descriptive or specific prompt\n",
+    "- Negative prompting, telling the model to ignore a particular thing i.e. the footnote date on every page\n",
```

Comment:
Not sure if negative prompting will work very well. Negative instructions are usually far less reliable than positive. I tried it, and the model still extracted a lot of 2022s.

Reply:
Removed negative prompting as a recommendation